app/models/taxonomy.rb

```diff
@@ -218,6 +218,17 @@ def notification_recipients_ids
     self.subtree.flat_map(&:users).map(&:id).uniq
   end
 
+  # note - this method used by before_destroy callbacks in extension files
+  # audits for 'destroy' action on resources lead to taxable_taxonomies records.
+  # This will check if any taxable_taxonomies records present and apply destroy_all
+  # so that it nullifies all associated audit records
+  def destroy_taxable_taxonomies_if_present
+    taxable_taxonomy_records = TaxableTaxonomy.where(taxonomy_id: self.id)
```

Comment:
I think the whole method could be simplified to
```
TaxableTaxonomy.where(taxonomy_id: self.id).destroy_all
```

if there are no records, it's safe to call destroy on that. If you agree, then I'd suggest to rename the method to `nullify_taxable_taxonomies` or `destroy_taxable_taxonomies` or something similar.

Reply:
Thank you @ares for review. 
Modified the code as per suggestions and updated the PR.